src/xenia/kernel/xfile.cc

```diff
@@ -122,14 +122,13 @@ X_STATUS XFile::Read(uint32_t buffer_guest_address, uint32_t buffer_length,
       const xe::BaseHeap* buffer_end_heap =
           memory()->LookupHeap(buffer_guest_high_address);
       if (!buffer_start_heap || !buffer_end_heap ||
-          buffer_start_heap->IsGuestPhysicalHeap() !=
-              buffer_end_heap->IsGuestPhysicalHeap() ||
-          (buffer_start_heap->IsGuestPhysicalHeap() &&
+          buffer_start_heap->heap_type() != buffer_end_heap->heap_type() ||
```

Comment:
This was specifically for disallowing access across the boundary of virtual and physical heaps, as there is special logic for  invalidating GPU memory in guest physical heaps (this isn't the correct behavior anyway from virtual memory perspective, but I made an assumption that this normally would result in an access violation anyway due to writing something near physical nullptr). While probably it doesn't make sense to touch XEX pages at all, the point in the original design was specifically to handle guest physical heaps — it's better to state intentions clearly than in a way that just happens to be work.